### PR TITLE
fix: Skip circluar references in examples

### PR DESCRIPTION
**Pull request by w-vi**

### CODE REVIEW

**Comment by pksunkara** [comment]
I am not sure if the fixture is actually getting tested. Also, can you please make the release commit in this PR too?

**Comment by w-vi** [comment]
The test gets tested : https://travis-ci.org/apiaryio/fury-adapter-swagger/builds/178621426#L534

**Comment by w-vi** [comment]
Comments addressed.

File: src/parser.js [hunk 1]
```diff
@@ -994,18 +994,22 @@ export default class Parser {
...
-              this.createSourceMap(bodyAsset, this.path);
-            }
+              if (this.generateSourceMap) {
+                this.createSourceMap(bodyAsset, this.path);
+              }
 
-            response.content.push(bodyAsset);
+              response.content.push(bodyAsset);
+            } catch (exception) {
+              this.createAnnotation(annotations.DATA_LOST,  this.path, 'Circular references in examples are not yet supported.');
```
Comment (kylef) [inline src/parser.js]: Wouldn't this catch other exceptions and falsy show a warning about circular references?

Comment (pksunkara) [inline src/parser.js]: Yeah, the `createSourceMap` might become an issue. Can you try catch only the JSON stringily please?

Comment (kylef) [inline src/parser.js]: Or perhaps we can filter for the specific exception we want to catch and re-raise the rest.

**Comment by w-vi** [comment]
Better?

**Pull request merged**